### DISCUSSION TITLE
Issues with routing wlan1 and wlan2, providing internet connectivity, when wlan0 is enabled

### DISCUSSION THREAD

**Discussion by matrixes** [RaspAP/raspap-webgui#1638] [Q&A]
# What I'm trying to achieve

Accessing the internet using two wifi dongles

# I have

- 1x RPi Zero 2 W
- 2x Wifi donglesl

# What I have done

I followed the instructions at https://docs.raspap.com/repeater/ with the key difference that I'm using wlan1 (AP) and wlan2 (client).

# What works

- I can connect to the RPi by wifi (wlan1).
- I can access RaspAP through the DHCP provided IP gateway address (wlan1).
- I can connect (wlan2) to a Wifi access point using RaspAP-webgui.
- The Networking tab is reporting that all connections are working, getting and providing appropriate IP:s
- If I disable/bring wlan0 down, routing between wlan1 and wlan2 works as I expect it to, providing internet connectivity

# What doesn't work

I have modified the metric on the DHCP tab of the DHCP Server settings for wlan1, but no matter if I leave it on default empty or a high metric, Internet is not routed.
If wlan0 is enabled (and e.g. gets an IP address), routing between wlan1 and wlan2 doesn't work as I expect it, providing no internet connectivity.

# Thoughts

I can certainly disable wlan0 and that would be an at least working setup. But are there other ways of keeping wlan0 running/enabled, and still have routing between wlan1 and wlan2 providing internet connectivity (preferrably using the webgui)?

**Comment by billz** [comment]
Impossible to determine without your routing table. If you have a working setup with wlan1 and wlan2, what's the purpose for keeping wlan0 enabled?

**Reply by matrixes** [reply]
My routing table as it currently stands

```
default via 10.3.141.1 dev wlan0 src 10.3.141.1 metric 3003
default via 192.168.0.1 dev wlan2 proto dhcp src 192.168.0.104 metric 3005
10.3.141.0/24 dev wlan0 proto dhcp scope link src 10.3.141.1 metric 3003
10.9.141.0/24 dev wlan1 proto dhcp scope link src 10.9.141.1 metric 3010
192.168.0.0/24 dev wlan0 proto kernel scope link src 192.168.0.132 metric 600
192.168.0.0/24 dev wlan2 proto dhcp scope link src 192.168.0.104 metric 3005
```

In response to purpose, I'd say it's mostly out of convenience and secondary that I'm curious whether it's possible to accomplish (_somewhat_ easily).

**Reply by billz** [reply]
Your wlan0 interface has multiple lower metric values, and therefore higher priorities, than your other interfaces. With this configuration, you will not have connectivity between wlan1 and wlan2. Review [how routing works](https://docs.raspap.com/repeater/#configure-routing) and correct this. 

Alternatively, you may delete these routes like so:

```
sudo ip route del default via 10.3.141.1 dev wlan0
sudo ip route del 192.168.0.0/24 dev wlan0
```

Note that the Linux kernel (not RaspAP) will add these routes again following a reboot. 

tl;dr: answered in the docs; please read them.
